### PR TITLE
Add version to bottom of settings

### PR DESCRIPTION
**Pull request by benalleng**
@futurepaul Happy to make it a full clickable settingsLink if you think it should live separately

### CODE REVIEW

File: src/components/layout/ExternalLink.tsx [hunk 1]
```diff
@@ -3,9 +3,9 @@ import { ParentComponent } from "solid-js";
 export const ExternalLink: ParentComponent<{ href: string }> = (props) => {
     return (
         <a target="_blank" rel="noopener noreferrer" href={props.href}>
-            {props.children}{" "}
+            {props.children}
             <svg
-                class="inline-block"
+                class="inline-block pl-0.5"
```
Comment (futurepaul) [inline src/components/layout/ExternalLink.tsx:8]: I don't think tailwind works like this. what were you trying to fix?

File: vite.config.ts [hunk 2]
```diff
@@ -49,6 +52,10 @@ export default defineConfig({
         }
     },
     plugins: [wasm(), solid({ ssr: false }), VitePWA(pwaOptions)],
+    define: {
+        "import.meta.env.__COMMIT_HASH__": JSON.stringify(commitHash),
+        "import.meta.env.__RELEASE_VERSION__": JSON.stringify("0.4.16")
```
Comment (futurepaul) [inline vite.config.ts:57]: looks like there's a pretty nice way to get this direct from package.json. one less thing to update every version bump! https://stackoverflow.com/questions/67194082/how-can-i-display-the-current-app-version-from-package-json-to-the-user-using-vi

**Review by futurepaul — CHANGES_REQUESTED** [review]
this is really slick. I think in the future we might want a whole "about" page where we can put more info but this is great for now. two comments on impl but concept ack

File: src/components/layout/ExternalLink.tsx (hunk 1 above)
Comment (benalleng) [reply src/components/layout/ExternalLink.tsx:8]: spacing between the text and the externalLink children and the upArrow, right now the whitespace string adds an underline over nothing.

image

image

File: vite.config.ts (hunk 2 above)
Comment (benalleng) [reply vite.config.ts:57]: well thats super easy

File: src/components/layout/ExternalLink.tsx (hunk 1 above)
Comment (benalleng) [reply src/components/layout/ExternalLink.tsx:8]: No theres a 0.5 https://tailwindcss.com/docs/padding

Comment (futurepaul) [reply src/components/layout/ExternalLink.tsx:8]: wow TIL I am in fact crazy

**Review by futurepaul — APPROVED** [review]
thank you!

**Pull request merged**